server/routes/visitJourney/additionalSupport.test.ts

```diff
@@ -46,379 +52,385 @@ afterEach(() => {
-        expect($('h1').text().trim()).toBe('Is additional support needed for any of the visitors?')
-        expect($('.govuk-error-summary__body').text()).toContain('No request selected')
-        expect($('[data-test="support-required-yes"]').prop('checked')).toBe(true)
-        expect($('#additionalSupport-error').text()).toContain('No request selected')
-        expect(flashProvider).toHaveBeenCalledWith('errors')
-        expect(flashProvider).toHaveBeenCalledWith('formValues')
-        expect(flashProvider).toHaveBeenCalledTimes(2)
-      })
-  })
+      flashData.formValues = [{ additionalSupportRequired: 'yes', additionalSupport: ['WHEELCHAIR', 'OTHER'] }]
 
-  it('should render validation errors from flash data when other support details not provided', () => {
-    flashData.errors = [
-      {
-        value: '',
-        msg: 'Enter details of the request',
-        param: 'otherSupportDetails',
-        location: 'body',
-      },
-    ]
-
-    flashData.formValues = [{ additionalSupportRequired: 'yes', additionalSupport: ['WHEELCHAIR', 'OTHER'] }]
-
-    return request(sessionApp)
-      .get('/book-a-visit/additional-support')
-      .expect(200)
-      .expect('Content-Type', /html/)
-      .expect(res => {
-        const $ = cheerio.load(res.text)
-        expect($('h1').text().trim()).toBe('Is additional support needed for any of the visitors?')
-        expect($('.govuk-error-summary__body').text()).toContain('Enter details of the request')
-        expect($('[data-test="support-required-yes"]').prop('checked')).toBe(true)
-        expect($('[data-test="WHEELCHAIR"]').prop('checked')).toBe(true)
-        expect($('[data-test="OTHER"]').prop('checked')).toBe(true)
-        expect($('#otherSupportDetails-error').text()).toContain('Enter details of the request')
-        expect(flashProvider).toHaveBeenCalledWith('errors')
-        expect(flashProvider).toHaveBeenCalledWith('formValues')
-        expect(flashProvider).toHaveBeenCalledTimes(2)
-      })
+      return request(sessionApp)
+        .get(`${journey.urlPrefix}/additional-support`)
+        .expect(200)
+        .expect('Content-Type', /html/)
+        .expect(res => {
+          const $ = cheerio.load(res.text)
+          expect($('h1').text().trim()).toBe('Is additional support needed for any of the visitors?')
+          expect($('.govuk-error-summary__body').text()).toContain('Enter details of the request')
+          expect($('[data-test="support-required-yes"]').prop('checked')).toBe(true)
+          expect($('[data-test="WHEELCHAIR"]').prop('checked')).toBe(true)
+          expect($('[data-test="OTHER"]').prop('checked')).toBe(true)
+          expect($('#otherSupportDetails-error').text()).toContain('Enter details of the request')
+          expect(flashProvider).toHaveBeenCalledWith('errors')
+          expect(flashProvider).toHaveBeenCalledWith('formValues')
+          expect(flashProvider).toHaveBeenCalledTimes(2)
+        })
+    })
   })
 })
```

Comment:
Not sure you need to end thee loop and restart on 235? Could just remove 233 and 235? Not tried but should work.

Reply:
I had tried it without and found the grouping was strange, will re-attempt though while I add the flag